### DISCUSSION TITLE
Why does the worded forecast in the JSON api differ from the worded forecast in the XML api and website?

### DISCUSSION THREAD

**Discussion by shaansheikh** [weather-gov/api#427] [Q&A] [answered]
I can get worded forecasts on the NWS website at
https://forecast.weather.gov/MapClick.php?lat={LAT}&lon={LON}
or via the XML api at
https://forecast.weather.gov/MapClick.php?lat={LAT}&lon={LON}&unit=0&lg=english&FcstType=dwml
where {LAT} and {LON} are the latitude and longitude.

I can also get the worded forecasts via the https://api.weather.gov/gridpoints/{wfo}/{x},{y}/forecast  endpoint.

However, the worded forecast on the site and xml api differs from the worded forecast in the JSON api for the same location. The site/xml worded forecast generally seems more detailed so I would prefer to get that one. Is there any way to get that forecast via the JSON api?

I notice the JSON api says `'forecastGenerator': 'BaselineForecastGenerator',` while the XML says ` wordGenerator="markMitchell"`. Is there any way to use the JSON api to get this `"markMitchell"` source for worded forecasts instead of `BaselineForecastGenerator`

**Comment by StephenClouse** [comment, marked as answer]
The code that generates text on the current forecast site is very old code (going back to the early 2000s) that was built in a way that we cannot integrate into the modern applications. API uses an entirely new codebase written from scratch. This is why the text is different between the two.

Our goal is to achieve parity between the two, but this is currently not our highest development priority. We will continue to make improvements to the API forecast generator as time allows.
